src/models/user.ts

```diff
@@ -160,6 +161,25 @@ export class User extends TypedEventEmitter<UserEvent, UserEventHandlerMap> {
         this.updateModifiedTime();
     }
 
+    /**
+     * Construct a new User whose events will also emit on MatrixClient.
+     * A User must have an ID and can optionally have extra information associated with it.
+     * @param userId - Required. The ID of this user.
+     * @param client - An instance of MatrixClient object
+     * @returns User object with reEmitter setup on client
+     */
+    public static withReEmitterOnClient(userId: string, client: MatrixClient): User {
```

Comment:
I think we should change the naming. It should not contain `withReEmitter`, but rather describe what it's doing.
The following seems more appropriate to me, and will survive a future refactor where we decide to move away from event emitters

```suggestion
    public static createUser(userId: string, client: MatrixClient): User {
```